app/models/ems_event.rb

```diff
@@ -25,7 +25,16 @@ def self.task_final_events
   end
 
   def self.event_groups
-    ::Settings.event_handling.event_groups.to_hash
+    core_event_groups = ::Settings.event_handling.event_groups.to_hash
+    Settings.ems.each_with_object(core_event_groups) do |(_provider_type, provider_settings), event_groups|
+      provider_event_groups = provider_settings.fetch_path(:event_handling, :event_groups)
+      next unless provider_event_groups
+      DeepMerge.deep_merge!(
+        provider_event_groups.to_hash, event_groups,
```

Comment:
nope, its `each_with_object(core_event_groups)` and `event_groups` is the memo object...